File: backend/app/schemas/agents.py
"""Pydantic schemas for agent endpoints."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class VoiceSelection(BaseModel):
    voice_id: str | None = None
    display_name: str | None = None
    provider_voice_id: str | None = None


class AgentResponse(BaseModel):
    id: str
    display_name: str
    function_type: str
    is_default: bool
    status: str
    voice: VoiceSelection | None = None
    user_instructions: str | None = None
    greeting_instructions: str | None = None
    revision: int = 1
    created_at: datetime

    model_config = {"from_attributes": True}


class AgentListResponse(BaseModel):
    items: list[AgentResponse]


class CreateDefaultAgentRequest(BaseModel):
    display_name: str | None = Field(None, max_length=60)
    voice_id: str | None = None
    user_instructions: str | None = Field(None, max_length=2000)


class UpdateAgentRequest(BaseModel):
    display_name: str | None = Field(None, max_length=60)
    voice_id: str | None = None
    user_instructions: str | None = Field(None, max_length=2000)
    greeting_instructions: str | None = Field(None, max_length=500)
    expected_revision: int | None = None


class VoiceCatalogItem(BaseModel):
    id: str
    provider_voice_id: str
    display_name: str
    locale: str
    gender_tag: str | None = None
    preview_url: str | None = None
    sort_order: int = 0

    model_config = {"from_attributes": True}


class VoiceCatalogResponse(BaseModel):
    items: list[VoiceCatalogItem]


class PromptSuggestionItem(BaseModel):
    id: str
    title: str
    text: str
    sort_order: int = 0

    model_config = {"from_attributes": True}


class PromptSuggestionsResponse(BaseModel):
    items: list[PromptSuggestionItem]


class AgentRuntimeResponse(BaseModel):
    """Internal-only response for Node bridge. Never exposed to mobile."""

    agent_id: str
    agent_display_name: str
    agent_function_type: str
    provider_voice_id: str
    final_prompt: str
    greeting_text: str = ""
    elevenlabs_agent_id: str = ""
    call_objective_mode: str = "screen_and_summarize"
    max_call_length_seconds: int = 180
    recording_enabled: bool = False
    memory_enabled: bool = True

File: mobile/src/api/agents.ts
import { apiClient } from './client';

export interface VoiceSelection {
  voice_id: string | null;
  display_name?: string | null;
  provider_voice_id?: string | null;
}

export interface AgentResponse {
  id: string;
  display_name: string;
  function_type: string;
  is_default: boolean;
  status: string;
  voice: VoiceSelection | null;
  user_instructions: string | null;
  greeting_instructions: string | null;
  revision: number;
  created_at: string;
}

export interface AgentListResponse {
  items: AgentResponse[];
}

export interface VoiceCatalogItem {
  id: string;
  provider_voice_id: string;
  display_name: string;
  locale: string;
  gender_tag: string | null;
  preview_url: string | null;
  sort_order: number;
}

export interface VoiceCatalogResponse {
  items: VoiceCatalogItem[];
}

export interface PromptSuggestionItem {
  id: string;
  title: string;
  text: string;
  sort_order: number;
}

export interface PromptSuggestionsResponse {
  items: PromptSuggestionItem[];
}

export async function fetchAgents(): Promise<AgentListResponse> {
  const { data } = await apiClient.get('/agents');
  return data;
}

export async function createDefaultAgent(body?: {
  display_name?: string;
  voice_id?: string;
  user_instructions?: string;
}): Promise<AgentResponse> {
  const { data } = await apiClient.post('/agents/default', body ?? {});
  return data;
}

export async function updateAgent(
  agentId: string,
  patch: {
    display_name?: string;
    voice_id?: string | null;
    user_instructions?: string | null;
    greeting_instructions?: string | null;
    expected_revision?: number;
  },
): Promise<AgentResponse> {
  const { data } = await apiClient.patch(`/agents/${agentId}`, patch);
  return data;
}

export async function fetchVoices(): Promise<VoiceCatalogResponse> {
  const { data } = await apiClient.get('/voices');
  return data;
}

export async function fetchPromptSuggestions(): Promise<PromptSuggestionsResponse> {
  const { data } = await apiClient.get('/prompt-suggestions');
  return data;
}
